clientcontroller/babylon.go

```diff
@@ -510,6 +510,66 @@ func (bc *BabylonController) QueryFinalityProviders() ([]*btcstakingtypes.Finali
 	return fps, nil
 }
 
+func (bc *BabylonController) QueryFinalityProvider(fpPk *btcec.PublicKey) (*btcstakingtypes.QueryFinalityProviderResponse, error) {
+	fpPubKey := bbntypes.NewBIP340PubKeyFromBTCPK(fpPk)
+	res, err := bc.bbnClient.QueryClient.FinalityProvider(fpPubKey.MarshalHex())
+	if err != nil {
+		return nil, fmt.Errorf("failed to query the finality provider %s: %v", fpPubKey.MarshalHex(), err)
+	}
+
+	return res, nil
+}
+
+func (bc *BabylonController) EditFinalityProviderDescription(fpPk *btcec.PublicKey,
+	reqDesc sttypes.Description) (*sttypes.Description, error) {
+
+	fpRes, err := bc.QueryFinalityProvider(fpPk)
```

Comment:
I think we should check `fpRes.Addr == bc.mustGetTxSigner()` as we require that only the one that controlls the babylon key of the finality provider can edit the fp

Reply:
Make sense, although this is checked on bbn, make sense to validate it here also to prevent unnecessary RPC